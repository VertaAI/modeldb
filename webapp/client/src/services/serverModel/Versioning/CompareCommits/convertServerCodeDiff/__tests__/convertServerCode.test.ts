import { convertServerCodeDiff } from '../index';

describe('convertServerCodeDiff', () => {
  describe('git', () => {
    it('should convert added diff', () => {
      const server = {
        location: ['repo_git'],
        status: 'ADDED',
        code: {
          git: {
            status: 'ADDED',
            B: {
              repo: 'repo',
              hash: '#hhh',
              branch: 'master',
              tag: 'tag',
              is_dirty: true,
            },
          },
        },
      };

      expect(convertServerCodeDiff(server as any)).toMatchSnapshot();
    });

    it('should convert deleted diff', () => {
      const server = {
        location: ['repo_git'],
        status: 'DELETED',
        code: {
          git: {
            status: 'DELETED',
            A: {
              repo: 'repo',
              hash: '#hhh',
              branch: 'master',
              tag: 'tag',
              is_dirty: true,
            },
          },
        },
      };

      expect(convertServerCodeDiff(server as any)).toMatchSnapshot();
    });

    it('should convert modified diff', () => {
      const server = {
        location: ['repo_gitadf'],
        status: 'MODIFIED',
        code: {
          git: {
            status: 'MODIFIED',
            A: {
              repo: 'repo',
              hash: '#hhh',
              branch: 'master',
              tag: 'tag',
              is_dirty: true,
            },
            B: {
              repo: 'repadfadfadfo',
              hash: '#hhh',
              branch: 'masteadfr',
              tag: 'tag',
              is_dirty: true,
            },
          },
        },
      };

      expect(convertServerCodeDiff(server as any)).toMatchSnapshot();
    });
  });
});
