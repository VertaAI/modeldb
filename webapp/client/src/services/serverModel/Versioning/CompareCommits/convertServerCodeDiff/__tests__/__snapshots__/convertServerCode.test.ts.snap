// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertServerCodeDiff git should convert added diff 1`] = `
Object {
  "category": "code",
  "data": Object {
    "B": Object {
      "data": Object {
        "branch": "master",
        "commitHash": "#hhh",
        "isDirty": true,
        "remoteRepoUrl": Object {
          "type": "unknown",
          "value": "repo",
        },
        "tag": "tag",
      },
      "type": "git",
    },
    "diffType": "added",
  },
  "diffType": "added",
  "location": Array [
    "repo_git",
  ],
  "type": "git",
}
`;

exports[`convertServerCodeDiff git should convert deleted diff 1`] = `
Object {
  "category": "code",
  "data": Object {
    "A": Object {
      "data": Object {
        "branch": "master",
        "commitHash": "#hhh",
        "isDirty": true,
        "remoteRepoUrl": Object {
          "type": "unknown",
          "value": "repo",
        },
        "tag": "tag",
      },
      "type": "git",
    },
    "diffType": "deleted",
  },
  "diffType": "deleted",
  "location": Array [
    "repo_git",
  ],
  "type": "git",
}
`;

exports[`convertServerCodeDiff git should convert modified diff 1`] = `
Object {
  "category": "code",
  "data": Object {
    "A": Object {
      "data": Object {
        "branch": "master",
        "commitHash": "#hhh",
        "isDirty": true,
        "remoteRepoUrl": Object {
          "type": "unknown",
          "value": "repo",
        },
        "tag": "tag",
      },
      "type": "git",
    },
    "B": Object {
      "data": Object {
        "branch": "masteadfr",
        "commitHash": "#hhh",
        "isDirty": true,
        "remoteRepoUrl": Object {
          "type": "unknown",
          "value": "repadfadfadfo",
        },
        "tag": "tag",
      },
      "type": "git",
    },
    "diffType": "modified",
  },
  "diffType": "modified",
  "location": Array [
    "repo_gitadf",
  ],
  "type": "git",
}
`;
